guides/authentication.md

```diff
@@ -212,7 +219,113 @@ client.loginWithOAuth({
 });
 ```
 
-## More
+### More
 
 Here is an article describing more in detail about the authentication process:
 https://doc.cognitedata.com/dev/guides/iam/external-application.html#tokens
+
+## Azure AD auth flow
+
+Azure AD auth flow can be used via the same `loginWithOAuth` method of `CogniteClient`,
+but different `options` structure has to be passed. You'll find what options structure has to be provided in examples below.
+Besides, here list of pre-requirements, that need to be in place to use Azure AD authentication flow:
+
+ - CDF project, which you're trying to log in, has to be configured to accept tokens issued by Azure AD (to be related to the cluster, that accepts Azure AD tokens, `bluefield` for instance)
+ - You should have the `clientId` of your Azure AD application token will be issued for
+ - Cluster your CDF project related to (`bluefield` for instance)
+
+Azure AD authentication flow supports two login types:
+
+ - [With redirect](#authentication-via-redirect)
+ - [With popup](#authentication-via-popup)
+
+### Authentication via redirect
+
+When you authenticate using redirect, auth flow remains quite similar that we have in CDF auth flow.
+But the IdP in this is Azure where user has to log in to prove identity.
+
+> You might find useful example application using redirect Azure AD auth flow [here](../samples/react/authentication-aad/src/App.js).
+> And don't forget to provde required environment variables via `.env` file.
+ 
+#### Redirect sign in type example
+
+```js
+import { CogniteClient } from '@cognite/sdk';
+const client = new CogniteClient({ ... });
+
+// tenantId can be also undefined,
+// microsoft /common endpoint will be used to define tenant instead
+client.loginWithOAuth({
+  cluster: 'cdf-cluster-name',
+  clientId: 'azure-application-client-id',
+  tenantId: 'azure-tenant-id'
+});
+
+// authenticate to the provided cluster
+await client.authenticate();
+
+// set project name to make requsts to
+client.setProject('cdf-project-name');
+
+// then use the SDK:
+const assets = await client.assets.retrive({ id: 23232789217132 });
+```
+
+With the call `await client.authenticate()` you'll be redirected to the IdP for log in.
```

Comment:
With the call `await client.authenticate()` you'll be redirected to the IdP to sign in.